pkg/process/procutil/process_linux.go

```diff
@@ -89,13 +96,54 @@ func (p *Probe) Close() {
 	}
 }
 
+// StatsByPID returns a map of stats info indexed by PID using the PIDs collected by ProcessesByPID()
+// return nil, nil if lastPIDs is empty
+func (p *Probe) StatsByPID(now time.Time) (map[int32]*Stats, error) {
```

Comment:
IMO this should take a slice of PIDs as an argument that are stored/managed elsewhere (the process-agent check itself) instead of maintaining the global state within procutil.